install.sh

```diff
@@ -23,41 +23,46 @@ elif [ -n "$TEA_SECRET" ]; then
     exit 1;;
   esac
 
-  if [ ! -f /usr/local/bin/tea ]; then
+  if test ! -f /usr/local/bin/tea; then
     tmp=$(mktemp -t tea.XXXXXXX)
-    curl --fail https://$TEA_SECRET/tea.xyz/$MIDFIX/v'*'.tar.gz -o $tmp
+    curl --fail https://"$TEA_SECRET"/tea.xyz/$MIDFIX/v'*'.tar.gz -o "$tmp"
     sudo mkdir -p /opt
+    # TODO: in Linux the answer is probably to chmod 777 /opt, unless it has a specific group
+    # we can add our user to (which isn't `root`)
     sudo chgrp staff /opt
     sudo chmod g+w /opt
     cd /opt
-    tar xzf $tmp
+    tar xzf "$tmp"
     cd tea.xyz
-    ln -sf $(ls -d */ | head -n 1) v'*'  #FIXME
+    if test ! -L v'*'; then
+      ln -sf "$(find . -maxdepth 1 -type d -name "v[0-9].[0-9].[0-9]" | head -n 1)" v'*'
+    fi
 
     sudo mkdir -p /usr/local/bin
     sudo ln -sf /opt/tea.xyz/v'*'/bin/tea /usr/local/bin/tea
 
     SHELLNAME=$(basename "$SHELL")
-    if [ "X$SHELLNAME" = "Xfish" ]
+    if test "$SHELLNAME" = "fish"
     then
-      echo "if type -q tea" > ~/.config/fish/conf.d/tea.fish
-      echo "  function __tea_env --on-variable PWD" >> ~/.config/fish/conf.d/tea.fish
-      echo "    eval (tea -Eds)" >> ~/.config/fish/conf.d/tea.fish
-      echo "  end" >> ~/.config/fish/conf.d/tea.fish
-      echo "end" >> ~/.config/fish/conf.d/tea.fish
+      {
+        echo "if type -q tea"
+        echo "  function __tea_env --on-variable PWD"
+        echo "    eval (tea -Eds)"
+        echo "  end"
+        echo "end"
+      } > ~/.config/fish/conf.d/tea.fish
     else
-      echo >> ~/.zshrc
-      echo '# added by tea' >> ~/.zshrc
-      echo 'add-zsh-hook -Uz chpwd (){ source <(tea -Eds) }' >> ~/.zshrc
+      {
+        echo
+        echo '# added by tea'
+        echo 'add-zsh-hook -Uz chpwd (){ source <(tea -Eds) }'
+      } >> ~/.zshrc
     fi
   fi
 
-  if [ "$#" -gt 1 ]; then
+  if test "$#" -gt 1; then
     exec /usr/local/bin/tea "$@"
   fi
 else
-  if test x$PAGER = x; then
-    PAGER=cat
-  fi
-  curl -Ssf https://raw.githubusercontent.com/teaxyz/white-paper/main/white-paper.md | $PAGER
+  curl -Ssf https://raw.githubusercontent.com/teaxyz/white-paper/main/white-paper.md | ${PAGER:-cat}
```

Comment:
I think I was trying to ensure POSIX compatibility here. But probs not needed

Reply:
I *believe* that _is_ POSIX-compliant.